portal-api/src/v1/users/login.controller.test.js

```diff
@@ -0,0 +1,199 @@
+    disabled: false,
+    hash: HASH,
+    salt: SALT,
+  };
+
+  it('returns the user and token when the user exists and the password is correct', async () => {
+    mockFindByUsernameSuccess(USER);
+    mockValidPasswordSuccess();
+    mockIssueJWTSuccess(USER);
+    mockUpdateLastLoginSuccess(USER);
+
+    const result = await login(USERNAME, PASSWORD);
+
+    expect(result).toEqual({ user: USER, tokenObject: TOKEN_OBJECT });
+  });
+
+  it("returns a 'userOrPasswordIncorrect' error when the user doesn't exist", async () => {
+    mockFindByUsernameReturnsNullUser();
+    mockValidPasswordSuccess();
+    mockIssueJWTSuccess(USER);
+    mockUpdateLastLoginSuccess(USER);
+
+    const result = await login(USERNAME, PASSWORD);
+
+    expect(result).toEqual({ error: userOrPasswordIncorrect });
+  });
+
+  it('returns an error if findByUsername returns an error message', async () => {
+    mockFindByUsernameReturnsError();
+    mockValidPasswordSuccess();
+    mockIssueJWTSuccess(USER);
+    mockUpdateLastLoginSuccess(USER);
+
+    const result = await login(USERNAME, PASSWORD);
+
+    expect(result).toEqual({ error: ERROR });
+  });
+
+  it("returns a 'userOrPasswordIncorrect' error when the password is incorrect", async () => {
+    mockFindByUsernameSuccess(USER);
+    mockValidPasswordFailure();
+    mockIssueJWTSuccess(USER);
+    mockUpdateLastLoginSuccess(USER);
+
+    const result = await login(USERNAME, PASSWORD);
+
+    expect(result).toEqual({ error: userOrPasswordIncorrect });
+  });
+
+  it("returns a 'userIsDisabled' error when the user is disabled", async () => {
+    const DISABLED_USER = { ...USER, disabled: true };
+
+    mockFindByUsernameSuccess(DISABLED_USER);
+    mockValidPasswordSuccess();
+    mockIssueJWTSuccess(DISABLED_USER);
+    mockUpdateLastLoginSuccess(DISABLED_USER);
+
+    const result = await login(USERNAME, PASSWORD);
+
+    expect(result).toEqual({ error: userIsDisabled });
+  });
+
+  it("returns a 'userIsBlocked' error when the user is blocked", async () => {
+    const BLOCKED_USER = { ...USER, 'user-status': 'blocked' };
+
+    mockFindByUsernameSuccess(BLOCKED_USER);
+    mockValidPasswordSuccess();
+    mockIssueJWTSuccess(BLOCKED_USER);
+    mockUpdateLastLoginSuccess(BLOCKED_USER);
+
+    const result = await login(USERNAME, PASSWORD);
+
+    expect(result).toEqual({ error: userIsBlocked });
+  });
+
+  it("returns a 'userOrPasswordIncorrect' error when the user doesn't exist and the user is disabled", async () => {
+    const DISABLED_USER = { ...USER, disabled: true };
+
+    mockFindByUsernameReturnsNullUser();
+    mockValidPasswordSuccess();
+    mockIssueJWTSuccess(DISABLED_USER);
+    mockUpdateLastLoginSuccess(DISABLED_USER);
+
+    const result = await login(USERNAME, PASSWORD);
+
+    expect(result).toEqual({ error: userOrPasswordIncorrect });
+  });
+
+  it("returns a 'userOrPasswordIncorrect' error when the user doesn't exist and the user is blocked", async () => {
```

Comment:
Similar to my previous comment in TFM, I'm a bit confused by these 2 scenarios where a user doesn't exist but is blocked / disabled
It's probably a test name wording thing :)